src/engine/BMInterface.php

```diff
@@ -432,6 +432,15 @@ public function save_join_game_decision($playerId, $gameId, $decision) {
         $game = $this->load_game($gameId);
 
         if (BMGameState::CHOOSE_JOIN_GAME != $game->gameState) {
+            if (('reject' == $decision) &&
+                ($playerId == $game->playerIdArray[0])) {
+                $decision = 'cancel';
+            }
+            $this->set_message(
+                'Your decision to ' .
+                $decision .
+                ' the game failed because the game has been updated in the meanwhile'
```

Comment:
Turning around pulls fast means i get to be super picky without feeling guilty about it.  Yay!

I don't think "in the meanwhile" is grammatical, and it doesn't really tell the player what's going on - which "meanwhile"?  How about "...the game has been updated since you loaded the page"?  Maybe that's too inside-baseball.